src/db/awsEs/locations-dao.js

```diff
@@ -92,32 +92,74 @@ const buildIdQueryBody = (queryParams) => {
   return esb.requestBodySearch().query(q).toJSON();
 };
 
+/**
+ * Generates a query body object to get all documents that match the given IDs
+ *
+ * @param {string[]} ids An array of document IDs to search for
+ * @param {string} type The IDs' document type. Should be either 'locations' or 'services'
+ * @returns {Object} Elasticsearch query body
+ */
+const buildBulkIdQueryBody = (ids, type) => {
+  const q = esb.boolQuery();
+  q.must(esb.idsQuery(type, ids));
+  return esb.requestBodySearch().query(q).toJSON();
+};
+
+/**
+ * Adds included services to raw Locations data rows
+ *
+ * @param {Object} res Response object from the GET Locations call
+ * @returns {Object} Returns the GET Locations response with the related services included
+ */
+const includeServices = async (res) => {
+  const client = Client(clientOptions());
+  await _.asyncEach(res.hits.hits, async ({ _source: locationSource }) => {
```

Comment:
Not sure, but can't we await until getting raw data from ES and them use normal lodash to iterate each of them?

Reply:
I'm not sure if we can get all of the included services before we need to associate them with the specific location. Alex and I also ran into an issue using the normal lodash function because the loop would return before any of the elasticsearch calls were made. I found this explaination from StackOverflow: `Using Babel will transform async/await to generator function and using forEach means that each iteration has an individual generator function, which has nothing to do with the others. so they will be executed independently and has no context of next() with others.` Maybe I'm misunderstanding your comment, but I think asyncEach is the best way to do this.

Reply:
Hmm, I think I'm a little confused now. Let's chat over Slack when you work today.

Reply:
In case you're curious @Ruefa , I just changed the logic of this function to work in the way the @tsoliangwu0130  suggested. Instead of using asyncEach to add the services to each location in series, we fire off all of the service promises and resolve them with Promise.all. Liang also mentioned there could be security concerns using async-dash, so I think this solution without the extra package is best.